python/ray/external_storage.py

```diff
@@ -243,6 +258,27 @@ def delete_spilled_objects(self, urls: List[str]):
             filename = parse_url_with_offset(url.decode()).base_url
             os.remove(os.path.join(self.directory_path, filename))
 
+    def destroy_external_storage(self):
+        # Q: Should we add stdout here to
+        # indicate we are deleting a directory?
+
+        # There's a race condition where IO workers are still
+        # deleting each objects while we try deleting the
+        # whole directory. So we should keep trying it until
+        # The directory is actually deleted.
+        while os.path.isdir(self.directory_path):
+            try:
+                shutil.rmtree(self.directory_path)
+            except FileNotFoundError:
+                # If excpetion occurs when other IO workers are
+                # deleting the file at the same time.
+                pass
+            except Exception:
+                logger.exception("Error cleaning up spill files\n"
+                                 f"Directory path: {self.directory_path}\n"
+                                 f"Traceback: {traceback.format_exc()}")
```

Comment:
You don't need to include the traceback; logger.exception already does this.

Reply:
Oh I see. Didn't know about that. I will fix this and merge the PR.